--- /home/jf/repos/lua/luatex_api_all/library/luatex/luaharfbuzz.lua_upstream.lua	2023-04-14 21:14:49.846846676 +0200
+++ /home/jf/repos/lua/luatex_api_all/library/luatex/luaharfbuzz.lua	2023-04-14 20:48:09.781665874 +0200
@@ -1,15 +1,15 @@
+---Changes to upstream: global luaharfbuzz table
+
 ---
 ---@meta
 ---The definitions are developed in this repository: https://github.com/LuaCATS/luaharfbuzz
 
+luaharfbuzz = {}
+
 ---
 ---https://github.com/ufyTeX/luaharfbuzz/blob/master/src/harfbuzz.luadoc
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
-local luaharfbuzz = {}
 
 ---
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 ---@class LuaHarfBuzzOptions
 ---@field direction Direction # A `Direction` object representing the object.
 ---@field script Script # A `Script` object representing the script.
@@ -25,24 +25,18 @@
 ---@param buffer Buffer
 ---@param features Feature[]
 ---@param shapers string[]
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function luaharfbuzz.shape_full(font, buffer, features, shapers) end
 
 ---
 ---* Corresponding C source code: [luaharfbuzz.c#L44-L53](https://github.com/ufyTeX/luaharfbuzz/blob/b3bdf5dc7a6e3f9b674226140c3dfdc73d2970cd/src/luaharfbuzz/luaharfbuzz.c#L44-L53)
 ---
 ---@return string ...  # for example: `graphite2 ot fallback`
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function luaharfbuzz.shapers() end
 
 ---
 ---Wraps `hb_version`
 ---
 ---* Corresponding C source code: [luaharfbuzz.c#L39-L42](https://github.com/ufyTeX/luaharfbuzz/blob/b3bdf5dc7a6e3f9b674226140c3dfdc73d2970cd/src/luaharfbuzz/luaharfbuzz.c#L39-L42)
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function luaharfbuzz.version() end
 
 ---
@@ -53,16 +47,12 @@
 ---@param font Font # `Font` to use for shaping
 ---@param buffer Buffer # `Buffer` to shape
 ---@param options? LuaHarfBuzzOptions
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function luaharfbuzz.shape(font, buffer, options) end
 
 ---
 ---Lua wrapper for `hb_blob_t` type
 ---
 ---@class Blob
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 local Blob = {}
 luaharfbuzz.Blob = Blob
 
@@ -75,8 +65,6 @@
 ---@param data string # lua string containing binary or character data.
 ---
 ---@return Blob
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function Blob.new(data) end
 
 ---
@@ -88,8 +76,6 @@
 ---@param filename string # lua string.
 ---
 ---@return Blob
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function Blob.new_from_file(filename) end
 
 ---
@@ -98,8 +84,6 @@
 ---* Corresponding C source code: [blob.c#L28-L33](https://github.com/ufyTeX/luaharfbuzz/blob/b3bdf5dc7a6e3f9b674226140c3dfdc73d2970cd/src/luaharfbuzz/blob.c#L28-L33)
 ---
 ---@return integer
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function Blob:get_length() end
 
 ---
@@ -108,14 +92,10 @@
 ---* Corresponding C source code: [blob.c#L35-L44](https://github.com/ufyTeX/luaharfbuzz/blob/b3bdf5dc7a6e3f9b674226140c3dfdc73d2970cd/src/luaharfbuzz/blob.c#L35-L44)
 ---
 ---@return string
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function Blob:get_data() end
 
 ---
 ---Lua wrapper for `hb_face_t` type
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 ---@class Face
 local Face = {}
 luaharfbuzz.Face = Face
@@ -130,8 +110,6 @@
 ---@param font_index? integer # index of font to read.
 ---
 ---@return Face|nil
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function Face.new_from_blob(blob, font_index) end
 
 ---
@@ -145,8 +123,6 @@
 ---@param font_index? integer # index of font to read.
 ---
 ---@return Face|nil
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function Face.new(file, font_index) end
 
 ---
@@ -155,14 +131,10 @@
 ---* Corresponding C source code: [face.c#L264-L284](https://github.com/ufyTeX/luaharfbuzz/blob/b3bdf5dc7a6e3f9b674226140c3dfdc73d2970cd/src/luaharfbuzz/face.c#L264-L284)
 ---
 ---@return table # of codepoints supported by the face.
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function Face:collect_unicodes() end
 
 ---
 ---Wraps `hb_face_get_glyph_count`.
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function Face:get_glyph_count() end
 
 ---
@@ -171,16 +143,12 @@
 ---@param tag Tag # `Tag` object of the table.
 ---
 ---@return Blob # `Blob` object for the face table of `tag`.
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function Face:get_table(tag) end
 
 ---
 ---Wraps `hb_face_get_table_tags`.
 ---
 ---@return Tag[] # table of `Tag`s representing face table tags.
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function Face:get_table_tags() end
 
 ---
@@ -189,8 +157,6 @@
 ---* Corresponding C source code: [face.c#L286-L291](https://github.com/ufyTeX/luaharfbuzz/blob/b3bdf5dc7a6e3f9b674226140c3dfdc73d2970cd/src/luaharfbuzz/face.c#L286-L291)
 ---
 ---@return integer
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function Face:get_upem() end
 
 ---
@@ -199,8 +165,6 @@
 ---* Corresponding C source code: [face.c#L293-L298](https://github.com/ufyTeX/luaharfbuzz/blob/b3bdf5dc7a6e3f9b674226140c3dfdc73d2970cd/src/luaharfbuzz/face.c#L293-L298)
 ---
 ---@return boolean
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function Face:ot_color_has_palettes() end
 
 ---
@@ -209,12 +173,9 @@
 ---* Corresponding C source code: [face.c#L300-L305](https://github.com/ufyTeX/luaharfbuzz/blob/b3bdf5dc7a6e3f9b674226140c3dfdc73d2970cd/src/luaharfbuzz/face.c#L300-L305)
 ---
 ---@return integer
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function Face:ot_color_palette_get_count() end
 
 ---
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 ---@class LuaHarfBuzzColor
 ---@field red integer
 ---@field green integer
@@ -227,8 +188,6 @@
 ---* Corresponding C source code: [face.c#L307-L347](https://github.com/ufyTeX/luaharfbuzz/blob/b3bdf5dc7a6e3f9b674226140c3dfdc73d2970cd/src/luaharfbuzz/face.c#L307-L347)
 ---
 ---@return LuaHarfBuzzColor[]
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function Face:ot_color_palette_get_colors() end
 
 ---
@@ -236,8 +195,6 @@
 ---
 ---* Corresponding C source code: [face.c#L349-L354](https://github.com/ufyTeX/luaharfbuzz/blob/b3bdf5dc7a6e3f9b674226140c3dfdc73d2970cd/src/luaharfbuzz/face.c#L349-L354)
 ---@return boolean
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function Face:ot_color_has_layers() end
 
 ---
@@ -245,8 +202,6 @@
 ---
 ---* Corresponding C source code: [face.c#L356-L392](https://github.com/ufyTeX/luaharfbuzz/blob/b3bdf5dc7a6e3f9b674226140c3dfdc73d2970cd/src/luaharfbuzz/face.c#L356-L392)
 ---
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function Face:ot_color_glyph_get_layers() end
 
 ---
@@ -255,8 +210,6 @@
 ---* Corresponding C source code: [face.c#L394-L399](https://github.com/ufyTeX/luaharfbuzz/blob/b3bdf5dc7a6e3f9b674226140c3dfdc73d2970cd/src/luaharfbuzz/face.c#L394-L399)
 ---
 ---@return boolean
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function Face:ot_color_has_png() end
 
 ---
@@ -265,8 +218,6 @@
 ---* Corresponding C source code: [face.c#L126-L156](https://github.com/ufyTeX/luaharfbuzz/blob/b3bdf5dc7a6e3f9b674226140c3dfdc73d2970cd/src/luaharfbuzz/face.c#L126-L156)
 ---
 ---@return Tag[]
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function Face:ot_layout_get_script_tags() end
 
 ---
@@ -275,8 +226,6 @@
 ---* Corresponding C source code: [face.c#L158-L189](https://github.com/ufyTeX/luaharfbuzz/blob/b3bdf5dc7a6e3f9b674226140c3dfdc73d2970cd/src/luaharfbuzz/face.c#L158-L189)
 ---
 ---@return Tag[]
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function Face:ot_layout_get_language_tags() end
 
 ---
@@ -285,8 +234,6 @@
 ---* Corresponding C source code: [face.c#L191-L223](https://github.com/ufyTeX/luaharfbuzz/blob/b3bdf5dc7a6e3f9b674226140c3dfdc73d2970cd/src/luaharfbuzz/face.c#L191-L223)
 ---
 ---@return Tag[]
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function Face:ot_layout_get_feature_tags() end
 
 ---
@@ -296,8 +243,6 @@
 ---
 ---@return boolean found
 ---@return integer index
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function Face:ot_layout_find_script() end
 
 ---
@@ -307,8 +252,6 @@
 ---
 ---@return boolean found
 ---@return integer index
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function Face:ot_layout_find_language() end
 
 ---
@@ -318,14 +261,11 @@
 ---
 ---@return boolean found
 ---@return integer index
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function Face:ot_layout_find_feature() end
 
 ---
 ---Lua wrapper for `hb_font_t` type
 ---
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 ---@class Font
 local Font = {}
 luaharfbuzz.Font = Font
@@ -337,8 +277,6 @@
 ---calling `hb_ot_font_set_funcs` on it.
 ---
 ---@param face Face # `Face` object.
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function Font.new(face) end
 
 ---
@@ -348,8 +286,6 @@
 ---
 ---@return integer # x-scale of the font.
 ---@return integer # y-scale of the font.
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function Font:get_scale() end
 
 ---
@@ -359,12 +295,9 @@
 ---
 ---@param x_scale integer # desired x-scale of font.
 ---@param y_scale integer # desired y-scale of font.
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function Font:set_scale(x_scale, y_scale) end
 
 ---
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 ---@class FontExtens
 ---@field ascender integer # typographic ascender.
 ---@field descender integer # typographic descender.
@@ -376,8 +309,6 @@
 ---@param glyph integer # index inside the font.
 ---
 ---@return FontExtens|nil # font extents table for horizontal direction, contains the following or `nil` if HarfBuzz fails to load font extents:
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function Font:get_h_extents(glyph) end
 
 ---
@@ -386,12 +317,9 @@
 ---@param glyph integer # index inside the font.
 ---
 ---@return FontExtens|nil # font extents table for vertical direction, similar to `Font:get_h_extents`, or `nil` if HarfBuzz fails to load font extents:
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function Font:get_v_extents(glyph) end
 
 ---
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 ---@class GlyphExtens
 ---@field x_bearing integer # left side of glyph from origin.
 ---@field y_bearing integer # top side of glyph from origin.
@@ -405,8 +333,6 @@
 ---
 ---@return GlyphExtens|nil # extents table contains the following or `nil` if HarfBuzz fails to
 ---load glyph extents
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function Font:get_glyph_extents(glyph) end
 
 ---
@@ -415,8 +341,6 @@
 ---@param glyph integer # index inside the font.
 ---
 ---@return string # name of the glyph or nil.
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function Font:get_glyph_name(glyph) end
 
 ---
@@ -425,8 +349,6 @@
 ---@param name string # name of the glyph.
 ---
 ---@return integer # glyph index inside the font or nil.
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function Font:get_glyph_from_name(name) end
 
 ---
@@ -435,8 +357,6 @@
 ---@param glyph integer # glyph index inside the font.
 ---
 ---@return integer # advance glyph advance of the glyph in horizontal direction.
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function Font:get_glyph_h_advance(glyph) end
 
 ---
@@ -445,8 +365,6 @@
 ---@param glyph integer # glyph index inside the font.
 ---
 ---@return integer # advance glyph advance of the glyph in vertical direction.
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function Font:get_glyph_v_advance(glyph) end
 
 ---
@@ -455,28 +373,20 @@
 ---@param codepoint integer
 ---
 ---@return integer|nil # glyph index or `nil` if `codepoint` is not supported by the font.
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function Font:get_nominal_glyph(codepoint) end
 
 ---
 ---Wraps `hb_ot_color_glyph_get_png`.
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function Font:ot_color_glyph_get_png() end
 
 ---
 ---Lua wrapper for `hb_buffer_t` type.
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 ---@class Buffer
 local Buffer = {}
 luaharfbuzz.Buffer = Buffer
 
 ---
 ---Wraps `hb_buffer_create`.
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function Buffer.new() end
 
 ---
@@ -484,8 +394,6 @@
 ---
 ---@param text string # UTF8 encoded string.
 ---@param opt? integer [opt=0] item_offset 0-indexed offset in `text`, from where to start adding. [opt=-1] item_length length to add from `item_offset`. `-1` adds till end of `text`.
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function Buffer:add_utf8(text, opt) end
 
 ---
@@ -493,97 +401,71 @@
 ---
 ---@param text integer[] # with codepoints as lua numbers.
 ---@param opt? integer [opt=0] item_offset 0-indexed offset in `text`, from where to start adding. [opt=-1] item_length length to add from `item_offset`. `-1` adds till end of `text`.
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function Buffer:add_codepoints(text, opt) end
 
 ---
 ---Wraps `hb_buffer_set_direction`.
 ---
 ---@param dir Direction # A `Direction` object.
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function Buffer:set_direction(dir) end
 
 ---
 ---Wraps `hb_buffer_get_direction`.
 ---
 ---@return Direction # A `Direction` object.
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function Buffer:get_direction() end
 
 ---
 ---Wraps `hb_buffer_set_script`.
 ---
 ---@param script Script # A `Script` object.
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function Buffer:set_script(script) end
 
 ---
 ---Wraps `hb_buffer_get_script`.
 ---
 ---@return Script # A `Script` object.
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function Buffer:get_script() end
 
 ---
 ---Wraps `hb_buffer_set_language`.
 ---
 ---@param lang Language A `Language` object
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function Buffer:set_language(lang) end
 
 ---
 ---Wraps `hb_buffer_get_language`.
 ---
 ---@return Language # A `Language` object
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function Buffer:get_language() end
 
 ---
 ---Wraps `hb_buffer_reverse`.
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function Buffer:reverse() end
 
 ---
 ---Wraps `hb_buffer_get_length`.
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function Buffer:get_length() end
 
 ---
 ---Wraps `hb_buffer_get_cluster_level`.
 ---
 ---@return integer level # see Cluster Levels
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function Buffer:get_cluster_level() end
 
 ---
 ---Wraps `hb_buffer_set_cluster_level`.
 ---
 ---@param level integer # see Cluster Levels
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function Buffer:set_cluster_level(level) end
 
 ---
 ---Wraps `hb_buffer_guess_segment_properties`.
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function Buffer:guess_segment_properties() end
 
 ---
 ---containing data for each glyph, in a nested table. Each nested
 ---table contains the following:
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 ---@class Glyph
 ---@field x_advance number # horizontal advance.
 ---@field y_advance number # vertical advance.
@@ -599,8 +481,6 @@
 ---`hb_glyph_info_get_glyph_flags`, and assembles the data into a Lua table.
 ---
 ---@return Glyph[]
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function Buffer:get_glyphs() end
 
 ---
@@ -609,45 +489,31 @@
 ---about what each of these levels mean.
 ---
 ---Wraps `HB_BUFFER_CLUSTER_LEVEL_MONOTONE_GRAPHEMES`.
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 Buffer.CLUSTER_LEVEL_MONOTONE_GRAPHEMES = 0
 
 ---
 ---Wraps `HB_BUFFER_CLUSTER_LEVEL_MONOTONE_CHARACTERS`.
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 Buffer.CLUSTER_LEVEL_MONOTONE_CHARACTERS = 1
 
 ---
 ---Wraps `HB_BUFFER_CLUSTER_LEVEL_CHARACTERS`.
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 Buffer.CLUSTER_LEVEL_CHARACTERS = 2
 
 ---
 ---Wraps `HB_BUFFER_CLUSTER_LEVEL_DEFAULT`.
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 Buffer.CLUSTER_LEVEL_DEFAULT = 0
 
 ---
 ---Wraps `HB_GLYPH_FLAG_UNSAFE_TO_BREAK`.
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 Buffer.GLYPH_FLAG_UNSAFE_TO_BREAK = 1
 
 ---
 ---Wraps `HB_GLYPH_FLAG_DEFINED`.
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 Buffer.GLYPH_FLAG_DEFINED = 3
 
 ---
 ---Lua wrapper for `hb_feature_t` type
 ---
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 ---@class Feature
 local Feature = {}
 luaharfbuzz.Feature = Feature
@@ -655,21 +521,16 @@
 ---
 ---Wraps `hb_feature_from_string`
 ---@param feature_string string # See [feature string syntax reference](https://github.com/ufytex/luaharfbuzz/wiki/Feature-Strings)
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function Feature.new(feature_string) end
 
 ---
 ---Wraps `hb_feature_to_string`.
 ---Enables nice output with `tostring(…)`.
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function Feature:__tostring() end
 
 ---
 ---Lua wrapper for `hb_tag_t` type.
 ---
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 ---@class Tag
 local Tag = {}
 luaharfbuzz.Tag = Tag
@@ -680,30 +541,23 @@
 ---@param s string # to be converted to a `Tag` object.
 ---
 ---@return Tag # a `Tag` object.
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function Tag.new(s) end
 
 ---
 ---Wraps `hb_tag_to_string`. Enable nice output with `tostring(…)`.
 ---
 ---@return string # Returns a string representation for the tag object.
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function Tag:__to_string() end
 
 ---
 ---Enables equality comparisions with `==` between two tags.
 ---
 ---@return boolean# `true` or `false` depending on whether the two tags are equal.
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function Tag:__eq() end
 
 ---
 ---Lua wrapper for `hb_script_t` type.
 ---
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 ---@class Script
 local Script = {}
 luaharfbuzz.Script = Script
@@ -714,8 +568,6 @@
 ---@param script string # 4-letter script code according to the [ISO 15924 standard](http://www.unicode.org/iso15924/iso15924-num.html).
 ---
 ---@return Script # a `Script` object.
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function Script.new(script) end
 
 ---
@@ -724,8 +576,6 @@
 ---* HarfBuzz online documentation: [harfbuzz-hb-common.html#hb-script-from-iso15924-tag](https://harfbuzz.github.io/harfbuzz-hb-common.html#hb-script-from-iso15924-tag)
 ---
 ---@param tag Tag # a `Tag` object representing a [ISO 15924 script](http://www.unicode.org/iso15924/iso15924-num.html).
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function Script.from_iso15924_tag(tag) end
 
 ---
@@ -734,61 +584,50 @@
 ---* HarfBuzz online documentation: [harfbuzz-hb-common.html#hb-script-to-iso15924-tag](https://harfbuzz.github.io/harfbuzz-hb-common.html#hb-script-to-iso15924-tag)
 ---
 ---@return Tag # a `Tag` object representing the script.
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function Script:to_iso15924_tag() end
 
 ---
 ---Enable nice output with `tostring(…)`
 ---
 ---@return string # Returns a 4-letter [ISO 15924 script code](http://www.unicode.org/iso15924/iso15924-num.html) for the script object.
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function Script:__to_string() end
 
 ---
 ---Enables equality comparisions with `==` between two scripts.
 ---@return boolean `true` or `false` depending on whether the two scripts are equal.
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function Script:__eq() end
 
 ---
+---Predefined Script Codes.
+---Predefined directions that correspond to their original definitions in Harfbuzz.
+---@section
+
+---
 ---Wraps `HB_SCRIPT_COMMON`.
 ---
 ---* [HarfBuzz online documentation](https://harfbuzz.github.io/harfbuzz-hb-common.html#HB-SCRIPT-COMMON:CAPS)
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 Script.COMMON = "Zyyy"
 
 ---
 ---Wraps `HB_SCRIPT_INHERITED`.
 ---
 ---* [HarfBuzz online documentation](https://harfbuzz.github.io/harfbuzz-hb-common.html#HB-SCRIPT-INHERITED:CAPS)
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 Script.INHERITED = "Zinh"
 
 ---
 ---Wraps `HB_SCRIPT_UNKNOWN`.
 ---
 ---* [HarfBuzz online documentation](https://harfbuzz.github.io/harfbuzz-hb-common.html#HB-SCRIPT-UNKNOWN:CAPS)
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 Script.UNKNOWN = "Zzzz"
 
 ---
 ---Wraps `HB_SCRIPT_INVALID`.
 ---
 ---* [HarfBuzz online documentation](https://harfbuzz.github.io/harfbuzz-hb-common.html#HB-SCRIPT-INVALID:CAPS)
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 Script.INVALID = "No script set"
 
 ---
 ---Lua wrapper for `hb_direction_t` type.
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 ---@class Direction
 local Direction = {}
 luaharfbuzz.Direction = Direction
@@ -798,64 +637,48 @@
 ---@param dir `ltr`|`rtl`|`ttb`|`btt`|`invalid` # can be one of `ltr`, `rtl`, `ttb`, `btt` or `invalid`.
 ---
 ---@return Direction # a `Direction` object.
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function Direction.new(dir) end
 
 ---
 ---Wraps `hb_direction_to_string`. Enable nice output with `tostring(…)`.
 ---
 ---@return string # Returns a string representation for direction.
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function Direction:__to_string() end
 
 ---
 ---Enables equality comparisions with `==` between two directions.
 ---
 ---@return boolean # `true` or `false` depending on whether the two tags are equal.
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function Direction:__eq() end
 
 ---
 ---Wraps `HB_DIRECTION_IS_VALID`.
 ---
 ---@return boolean # a boolean value
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function Direction:is_valid() end
 
 ---
 ---Wraps `HB_DIRECTION_IS_HORIZONTAL`.
 ---
 ---@return boolean # a boolean value
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function Direction:is_horizontal() end
 
 ---
 ---Wraps `HB_DIRECTION_IS_VERTICAL`.
 ---
 ---@return boolean # a boolean value
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function Direction:is_vertical() end
 
 ---
 ---Wraps `HB_DIRECTION_IS_FORWARD`.
 ---
 ---@return boolean # a boolean value
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function Direction:is_forward() end
 
 ---
 ---Wraps `HB_DIRECTION_IS_BACKWARD`.
 ---
 ---@return boolean # a boolean value
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function Direction:is_backward() end
 
 ---
@@ -863,8 +686,6 @@
 ---
 ---Predefined directions that correspond to their original definitions in Harfbuzz.
 ---@type integer|nil
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 luaharfbuzz.Direction.LTR = 4
 
 ---
@@ -872,8 +693,6 @@
 ---
 ---Predefined directions that correspond to their original definitions in Harfbuzz.
 ---@type integer|nil
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 luaharfbuzz.Direction.RTL = 0
 
 ---
@@ -881,8 +700,6 @@
 ---
 ---Predefined directions that correspond to their original definitions in Harfbuzz.
 ---@type integer|nil
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 luaharfbuzz.Direction.TTB = 0
 
 ---
@@ -890,40 +707,30 @@
 ---
 ---Predefined directions that correspond to their original definitions in Harfbuzz.
 ---@type integer|nil
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 luaharfbuzz.Direction.BTT = 0
 
 ---
 ---Lua wrapper for `hb_language_t` type.
 ---@class Language
 local Language = {}
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 luaharfbuzz.Language = Language
 
 ---
 ---Wraps `hb_language_from_string`.
 ---@param lang string [three-letter language tag](http://www.microsoft.com/typography/otspec/languagetags.htm) to be converted to a `Language` object.
 ---@return Language # a `Language` object.
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function Language.new(lang) end
 
 ---
 ---Wraps `hb_language_to_string`. Enable nice output with `tostring(…)`.
 ---
 ---@return string # Returns a string representation for the language object.
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function Language:__to_string() end
 
 ---
 ---Enables equality comparisions with `==` between two languages.
 ---
 ---@return boolean # `true` or `false` depending on whether the two languages are equal.
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function Language:__eq() end
 
 ---
@@ -940,15 +747,11 @@
 ---Wraps `hb_unicode_script`
 ---@param char integer # Unicode codepoint
 ---@return Script # a `Script` object.
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 function unicode.script(char) end
 
 ---
 ---Predefined Name IDs.
 ---Predefined OpenType 'name' table name identifier.
----
----😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
 ---@class ot
 ---@field NAME_ID_COPYRIGHT string # Wraps `HB_OT_NAME_ID_COPYRIGHT`
 ---@field NAME_ID_FONT_FAMILY string # Wraps `HB_OT_NAME_ID_FONT_FAMILY`
