--- /home/jf/repos/lua/luatex_api_all/library/luatex/luaharfbuzz.lua_upstream.lua	2023-04-17 21:27:10.120411690 +0200
+++ /home/jf/repos/lua/luatex_api_all/library/luatex/luaharfbuzz.lua	2023-04-17 21:27:04.075902524 +0200
@@ -1,3 +1,8 @@
+_N._13_1_The_luaharfbuzz_library = "page 255"
+
+---
+---Changes to upstream: global luaharfbuzz table
+
 ---
 ---@meta
 ---The definitions are developed in this repository: https://github.com/LuaCATS/luaharfbuzz
@@ -6,7 +11,7 @@
 ---https://github.com/ufyTeX/luaharfbuzz/blob/master/src/harfbuzz.luadoc
 ---
 ---😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
-local luaharfbuzz = {}
+luaharfbuzz = {}

 ---
 ---😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
