--- /home/jf/repos/lua/luatex_api_all/library/luatex/luaharfbuzz.lua_upstream.lua	2023-04-14 22:11:30.093750138 +0200
+++ /home/jf/repos/lua/luatex_api_all/library/luatex/luaharfbuzz.lua	2023-04-14 22:07:38.078394498 +0200
@@ -1,3 +1,5 @@
+---Changes to upstream: global luaharfbuzz table
+
 ---
 ---@meta
 ---The definitions are developed in this repository: https://github.com/LuaCATS/luaharfbuzz
@@ -6,7 +8,7 @@
 ---https://github.com/ufyTeX/luaharfbuzz/blob/master/src/harfbuzz.luadoc
 ---
 ---😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
-local luaharfbuzz = {}
+luaharfbuzz = {}
 
 ---
 ---😱 [Types](https://github.com/LuaCATS/luaharfbuzz/blob/main/library/luaharfbuzz.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luaharfbuzz/pulls)
