--- /home/jf/repos/lua/luatex_api_all/library/luatex/mime.lua_upstream.lua	2023-04-14 22:11:30.541779757 +0200
+++ /home/jf/repos/lua/luatex_api_all/library/luatex/mime.lua	2023-04-14 22:09:11.664567056 +0200
@@ -1,3 +1,8 @@
+_N._4_3_lua_modules = "page 67"
+
+---
+---https://github.com/TeX-Live/luatex/blob/trunk/source/texk/web2c/luatexdir/luasocket/src/mime.lua
+---Changes to upstream: global mime table
 ---@meta
 ---The definitions are developed in this repository: https://github.com/LuaCATS/luasocket
 
@@ -28,7 +33,7 @@
 ---```
 ---
 ---😱 [Types](https://github.com/LuaCATS/luasocket/blob/main/library/mime.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luasocket/pulls)
-local mime = {}
+mime = {}
 
 ---
 ---Returns a filter that decodes data from a given transfer content
