--- /home/jf/repos/lua/luatex_api_all/library/luatex/unicode.lua_upstream.lua	2023-04-17 08:40:18.291040709 +0200
+++ /home/jf/repos/lua/luatex_api_all/library/luatex/unicode.lua	2023-04-17 08:40:02.035280999 +0200
@@ -1,3 +1,14 @@
+_N._4_3_lua_modules = "page 67"
+
+---
+---`slnunicode`, from the `selene` libraries, http://luaforge.net/projects/sln. This library has been slightly extended
+---so that the `unicode.utf8.*` functions also accept the first 256 values
+---of plane 18. This is the range *LuaTeX* uses for raw binary output, as
+---explained above. We have no plans to provide more like this because you can
+---basically do all that you want in *Lua*.
+---
+---Changes to the upstream project: global unicode table
+
 ---
 ---@meta
 ---The definitions are developed in this repository: https://github.com/LuaCATS/slnunicode
@@ -22,7 +33,7 @@
 ---"grapheme extension" characters (Mn+Me) like combining diacritical marks.
 ---
 ---😱 [Types](https://github.com/LuaCATS/slnunicode/blob/main/library/unicode.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/slnunicode/pulls)
-local unicode = {}
+unicode = {}

 ---
 ---😱 [Types](https://github.com/LuaCATS/slnunicode/blob/main/library/unicode.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/slnunicode/pulls)
