--- /home/jf/repos/lua/luatex_api_all/library/luatex/zlib.lua_upstream.lua	2023-04-14 21:08:15.596477445 +0200
+++ /home/jf/repos/lua/luatex_api_all/library/luatex/zlib.lua	2023-04-14 20:48:09.773665971 +0200
@@ -1,3 +1,7 @@
+_N._4_3_lua_modules = "page 67"
+---Changes to upstream: global zlib table
+
+---
 ---@meta
 ---The definitions are developed in this repository: https://github.com/LuaCATS/lzlib
 
@@ -5,16 +9,12 @@
 ---`lzlib`, by Tiago Dionizio, http://luaforge.net/projects/lzlib/.
 ---https://luarocks.org/modules/hisham/lzlib
 ---https://github.com/LuaDist/lzlib
----
----😱 [Types](https://github.com/LuaCATS/lzlib/blob/main/library/lzlib.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/lzlib/pulls)
-local zlib = {}
+zlib = {}
 
 ---
 ---returns zlib version
 ---
 ---@return string
----
----😱 [Types](https://github.com/LuaCATS/lzlib/blob/main/library/lzlib.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/lzlib/pulls)
 function zlib.version() end
 
 ---
@@ -27,8 +27,6 @@
 ---@param buffer? string
 ---
 ---@return string buffer
----
----😱 [Types](https://github.com/LuaCATS/lzlib/blob/main/library/lzlib.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/lzlib/pulls)
 function zlib.adler32(adler32, buffer) end
 
 ---
@@ -38,8 +36,6 @@
 ---@param buffer? string
 ---
 ---@return string buffer
----
----😱 [Types](https://github.com/LuaCATS/lzlib/blob/main/library/lzlib.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/lzlib/pulls)
 function zlib.crc32(crc32, buffer) end
 
 ---
@@ -53,8 +49,6 @@
 ---@param strategy? integer
 ---
 ---@return string buffer
----
----😱 [Types](https://github.com/LuaCATS/lzlib/blob/main/library/lzlib.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/lzlib/pulls)
 function zlib.compress(buffer, level, method, windowBits, memLevel, strategy) end
 
 ---
@@ -64,25 +58,14 @@
 ---@param windowBits? integer
 ---
 ---@return string
----
----😱 [Types](https://github.com/LuaCATS/lzlib/blob/main/library/lzlib.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/lzlib/pulls)
 function zlib.decompress(buffer, windowBits) end
 
 ---
----😱 [Types](https://github.com/LuaCATS/lzlib/blob/main/library/lzlib.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/lzlib/pulls)
 ---@class Sink
 local Sink = {}
 
----
----😱 [Types](https://github.com/LuaCATS/lzlib/blob/main/library/lzlib.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/lzlib/pulls)
 function Sink:write() end
-
----
----😱 [Types](https://github.com/LuaCATS/lzlib/blob/main/library/lzlib.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/lzlib/pulls)
 function Sink:close() end
-
----
----😱 [Types](https://github.com/LuaCATS/lzlib/blob/main/library/lzlib.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/lzlib/pulls)
 function Sink:flush() end
 
 ---
@@ -97,19 +80,12 @@
 ---@param dictionary? string - default `""`
 ---
 ---@return Stream
----
----😱 [Types](https://github.com/LuaCATS/lzlib/blob/main/library/lzlib.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/lzlib/pulls)
 function zlib.deflate(sink, level, method, windowBits, memLevel, strategy, dictionary) end
 
 ---
----😱 [Types](https://github.com/LuaCATS/lzlib/blob/main/library/lzlib.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/lzlib/pulls)
 ---@class InflateSink
 local InflateSink = {}
----
----😱 [Types](https://github.com/LuaCATS/lzlib/blob/main/library/lzlib.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/lzlib/pulls)
 function InflateSink:read() end
----
----😱 [Types](https://github.com/LuaCATS/lzlib/blob/main/library/lzlib.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/lzlib/pulls)
 function InflateSink:close() end
 
 ---
@@ -119,12 +95,9 @@
 ---@param windowBits? integer -  default `15`
 ---@param dictionary? string - default `""`
 ---@return Stream
----
----😱 [Types](https://github.com/LuaCATS/lzlib/blob/main/library/lzlib.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/lzlib/pulls)
 function zlib.inflate(source, windowBits, dictionary) end
 
 ---
----😱 [Types](https://github.com/LuaCATS/lzlib/blob/main/library/lzlib.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/lzlib/pulls)
 ---@class Stream
 local Stream = {}
 
@@ -132,8 +105,6 @@
 ---Write each parameter into the stream.
 ---
 ---@param ... any
----
----😱 [Types](https://github.com/LuaCATS/lzlib/blob/main/library/lzlib.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/lzlib/pulls)
 function Stream:write(...) end
 
 ---
@@ -146,29 +117,21 @@
 ---    'a' - reads the remaining bytes
 ---    'l' - reads a line
 ---@param ... integer|'a'|'l'
----
----😱 [Types](https://github.com/LuaCATS/lzlib/blob/main/library/lzlib.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/lzlib/pulls)
 function Stream:read(...) end
 
 ---
 ---Returns an iterator that returns a new line each time
 ---it is called.
 ---@return function
----
----😱 [Types](https://github.com/LuaCATS/lzlib/blob/main/library/lzlib.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/lzlib/pulls)
 function Stream:lines() end
 
 ---
 ---Flush output for deflate streams.
 ---@param opts 'sync'|'full'|'finish'
----
----😱 [Types](https://github.com/LuaCATS/lzlib/blob/main/library/lzlib.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/lzlib/pulls)
 function Stream:flush(opts) end
 
 ---
 ---Close the stream.
----
----😱 [Types](https://github.com/LuaCATS/lzlib/blob/main/library/lzlib.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/lzlib/pulls)
 function Stream:close() end
 
 return zlib
