--- /home/jf/repos/lua/luatex_api_all/library/luatex/zlib.lua_upstream.lua	2023-04-17 21:44:13.293592373 +0200
+++ /home/jf/repos/lua/luatex_api_all/library/luatex/zlib.lua	2023-04-17 21:44:08.949322763 +0200
@@ -1,3 +1,8 @@
+_N._4_3_lua_modules = "page 67"
+
+---
+---Changes to upstream: global zlib table
+
 ---
 ---@meta
 ---The definitions are developed in this repository: https://github.com/LuaCATS/lzlib
