--- /home/jf/repos/lua/luatex_api_all/library/luatex/mbox.lua_upstream.lua	2023-04-20 11:21:42.013515405 +0200
+++ /home/jf/repos/lua/luatex_api_all/library/luatex/mbox.lua	2023-04-20 11:21:23.583779644 +0200
@@ -1,3 +1,9 @@
+_N._4_3_lua_modules = "page 67"
+
+---
+---https://gitlab.lisn.upsaclay.fr/texlive/luatex/-/blob/trunk/source/texk/web2c/luatexdir/luasocket/src/mbox.lua
+---Changes to upstream: global mbox table
+
 ---
 ---@meta
 ---The definitions are developed in this repository: https://github.com/LuaCATS/luasocket
@@ -6,7 +12,7 @@
 ---https://github.com/lunarmodules/luasocket/blob/master/src/mbox.lua
 ---
 ---😱 [Types](https://github.com/LuaCATS/luasocket/blob/main/library/mbox.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/luasocket/pulls)
-local mbox = {}
+mbox = {}

 ---
 ---@param message_s string
