--- /home/jf/repos/lua/luatex_api_all/library/luatex/zlib.lua_upstream.lua	2023-04-19 22:07:10.247100973 +0200
+++ /home/jf/repos/lua/luatex_api_all/library/luatex/zlib.lua	2023-04-19 17:03:49.350909019 +0200
@@ -1,3 +1,8 @@
+_N._4_3_lua_modules = "page 67"
+
+---
+---Changes to upstream: global zlib table
+
 ---
 ---@meta
 ---The definitions are developed in this repository: https://github.com/LuaCATS/lzlib
@@ -10,7 +15,7 @@
 ---* On luarocks: https://luarocks.org/modules/hisham/lzlib
 ---
 ---😱 [Types](https://github.com/LuaCATS/lzlib/blob/main/library/lzlib.lua) incomplete or incorrect? 🙏 [Please contribute!](https://github.com/LuaCATS/lzlib/pulls)
-local zlib = {}
+zlib = {}

 ---
 ---Return the zlib version.
